### PR TITLE
feat: profile page, remove old http-client, create fetcher, refresh token via middleware

### PR DESCRIPTION
**Pull request by ndt080**

### CODE REVIEW

**Comment by ndt080** [comment]
quite a lot of file deletions and relocations, so there are so many changes

File: src/entities/user/api/useUserApi.ts [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
-  const $http = withHttpClient();
+export function useUserApi(cookies?: () => RequestCookies | ReadonlyRequestCookies) {
+  const accessToken = isServer()
+    ? cookies?.().get(StorageKey.AccessToken)?.value ?? ''
+    : (getCookie(StorageKey.AccessToken) as string) ?? '';
 
   const fetchCurrentUser = (): Promise<User> => {
-    return $http.get<ApiUser>(ENDPOINTS.CURRENT_USER, { signRequest: true }).then(UserMapper.map);
+    const headers = createFetcherHeaders({ Authorization: `Bearer ${accessToken}` });
+    const options = { headers, cache: 'no-store' as RequestCache };
```
Comment (kirillkurko) [inline src/entities/user/api/useUserApi.ts]: How about specifying options object type (should be `RequestInit`) so we don't use `as`

File: src/entities/user/api/useUserApi.ts [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+    return fetcher<ApiUser>('get', url(ENDPOINTS.CURRENT_USER), options).then(UserMapper.map);
   };
 
   const updateUsername = (username: string): Promise<User> => {
-    return $http
-      .patch<ApiUser>(ENDPOINTS.CURRENT_USER, { username }, { signRequest: true })
-      .then(UserMapper.map);
+    const headers = createFetcherHeaders({ Authorization: `Bearer ${accessToken}` });
+    const body = JSON.stringify({ username });
+    const options = { body, headers, cache: 'no-store' as RequestCache };
```
Comment (kirillkurko) [inline src/entities/user/api/useUserApi.ts]: How about specifying options object type (should be RequestInit) so we don't use as

File: src/shared/ui/Avatar/utils/generateInitials.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export function generateInitials(value = '') {
+  return value
+    .split(' ')
+    .slice(0, 2)
```
Comment (kirillkurko) [inline src/shared/ui/Avatar/utils/generateInitials.ts:4]: Do we really need slice here? I assume initials should be generated from a string following "Name Surname" pattern (so we don't need `slice(0, 2)`), but I might be missing something

**Review by kirillkurko — APPROVED** [review]
Nice! 🔥 Just a few small comments, but feel free to merge this one. 

I really like your project structure 🚀

File: src/shared/ui/Avatar/utils/generateInitials.ts (hunk 3 above)
Comment (ndt080) [reply src/shared/ui/Avatar/utils/generateInitials.ts:4]: The username can have many spaces (not just one, there are no restrictions), and we need a maximum of 2 letters.

**Pull request merged**